OpenRA.Mods.Common/UtilityCommands/UpgradeRules.cs

```diff
@@ -934,7 +934,144 @@ internal static void UpgradeActors(int engineVersion, ref List<MiniYamlNode> nod
 		{
 			foreach (var node in nodes)
 			{
-				// Add rules here
+				// Fix RA building footprints to not use _ when it's not necessary
+				if (engineVersion < 20160530 && Game.ModData.Manifest.Mod.Id == "ra")
+				{
+					// Tesla Coil
+					if (node.Value.Value == "tsla")
```

Comment:
This is for upgrading maps from the resource center that don't ship with OpenRA, mostly.
@pchote requested them (iirc).